snapcraft/tests/fixture_setup.py

```diff
@@ -90,6 +91,32 @@ def setUp(self):
         self.addCleanup(patcher_dirs.stop)
 
 
+class FakeProjectOptions(fixtures.Fixture):
+
+    def __init__(self, **kwargs):
+        self._kwargs = dict(
+            arch_triplet=kwargs.pop('arch_triplet', 'x86_64-gnu-linux'),
+            parts_dir=kwargs.pop('parts_dir', 'parts'),
+            stage_dir=kwargs.pop('stage_dir', 'stage'),
+            prime_dir=kwargs.pop('prime_dir', 'prime'),
+            parallel_build_count=kwargs.pop('parallel_build_count', '1'),
+        )
+        if kwargs:
+            raise NotImplementedError(
+                'Handling of {!r} is not implemented'.format(kwargs.keys()))
+
+    def setUp(self):
+        super().setUp()
+
+        patcher = mock.patch('snapcraft.ProjectOptions')
+        patcher.start()
+        self.addCleanup(patcher.stop)
+
+        project_options_t = type(snapcraft.ProjectOptions.return_value)
+        for key in self._kwargs:
+            setattr(project_options_t, key, self._kwargs[key])
```

Comment:
I'm not sure I'm understanding this. You get the type of the return value, which is a mock?
I might be misunderstanding your idea, but a named tuple can be sued to convert the kwargs dictionary into an object. Also, the kwargs in a Mock are then turned into attributes:

>>> a = mock.Mock(test='a')
>>> a.test
'a'

Maybe one of those can make this code clearer.